R/waapwlspetpeese.R

```diff
@@ -0,0 +1,610 @@
+      se      = models[[estimator]][toupper(estimator), "Std. Error"],
+      stat    = models[[estimator]][toupper(estimator), "t value"],
+      df      = attr(models[[estimator]], "df"),
+      pVal    = models[[estimator]][toupper(estimator), "Pr(>|t|)"],
+      lowerCI = models[[estimator]][toupper(estimator), "lCI"],
+      upperCI = models[[estimator]][toupper(estimator), "uCI"]
+    ))
+  }
+
+  return(table)
+}
+.wwppFillHeterogeneity       <- function(jaspResults, table, models, options, type) {
+
+  overtitleCI <- gettextf("95%% Confidence Interval")
+
+  table$addColumnInfo(name = "type",     title = "",                        type = "string")
+  table$addColumnInfo(name = "est",      title = gettext("Estimate"),       type = "number")
+
+
+  if (is.null(models))
+    return(table)
+
+
+  for (estimator in if(type == "waapWls") c("wls", "waap") else c("pet", "peese")) {
+
+    if (length(models[[estimator]]) == 0 && estimator == "waap") {
+      table$addFootnote(symbol = gettextf("Error: There was not enough adequatelly powered studies (k = %1$i)", attr(models[["waap"]], "nPowered")))
+    } else {
+      table$addRows(list(
+        type    = toupper(estimator),
+        est     = attr(models[[estimator]], "sigma")
+      ))
+      if (estimator == "waap")
+        table$addFootnote(symbol = gettextf("Note: There were %1$i adequatelly powered studies", attr(models[["waap"]], "nPowered")))
+    }
+  }
+
+  return(table)
+}
+# fit models (the fitting functions are adapted from Carter et al., 2019)
+.wwppFit                     <- function(jaspResults, dataset, options, type) {
+
+  if (!is.null(jaspResults[["models"]])) {
+    return()
+  } else {
+    models <- createJaspState()
+    models$dependOn(c(.wwppDependencies, "tablePVal"))
+    jaspResults[["models"]] <- models
+  }
+
+  if (type == "waapWls") {
+
+    summaryWls   <- .wwppWls(  .maGetInputEs(dataset, options), .maGetInputSe(dataset, options))
+    summaryWaap  <- .wwppWaap( .maGetInputEs(dataset, options), .maGetInputSe(dataset, options))
+
+    # take care of the transformed estimates
+    if (options[["measures"]] == "correlation") {
+      summaryWls   <- .wwppTransformEstimates(summaryWls,   options)
+      summaryWaap  <- .wwppTransformEstimates(summaryWaap,  options)
+    }
+
+    models[["object"]] <- list(
+      wls   = summaryWls,
+      waap  = summaryWaap
+    )
+
+  } else if (type == "petPeese") {
+
+    summaryPet   <- .wwppPet(  .maGetInputEs(dataset, options), .maGetInputSe(dataset, options))
+    summaryPeese <- .wwppPeese(.maGetInputEs(dataset, options), .maGetInputSe(dataset, options))
+
+    # take care of the transformed estimates
+    if (options[["measures"]] == "correlation") {
+      summaryPet   <- .wwppTransformEstimates(summaryPet,   options)
+      summaryPeese <- .wwppTransformEstimates(summaryPeese, options)
+    }
+
+    models[["object"]] <- list(
+      pet   = summaryPet,
+      peese = summaryPeese
+    )
+
+  }
+
+  return()
```

Comment:
The `return()` is not necessary right?

Reply:
no, but I thought that it was preferred to write them explicitly